packages/ember-data/lib/system/model/model.js

```diff
@@ -781,7 +782,11 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   hasMany: function() {
     Ember.assert("The `hasMany` method is not available on DS.Model, a DS.Snapshot was probably expected. Are you passing a DS.Model instead of a DS.Snapshot to your serializer?", false);
-  }
+  },
+
+  setId: Ember.observer('id', function () {
```

Comment:
But what if someone needs to edit the id before saving a new model? That worked before 1.0.0-beta.19